File: src/main/java/com/chidozie/core/shadows/ShadowShader.java
package com.chidozie.core.shadows;

import com.chidozie.core.shaders.ShaderProgram;
import org.joml.Matrix4f;

public class ShadowShader extends ShaderProgram {
	
	private static final String VERTEX_FILE = "shadows/shadowVertexShader.vert";
	private static final String FRAGMENT_FILE = "shadows/shadowFragmentShader.frag";
	
	private int location_mvpMatrix;

	protected ShadowShader() {
		super(VERTEX_FILE, FRAGMENT_FILE);
	}

	@Override
	protected void getAllUniformLocations() {
		location_mvpMatrix = super.getUniformLocation("mvpMatrix");
		
	}
	
	protected void loadMvpMatrix(Matrix4f mvpMatrix){
		super.loadMatrix(location_mvpMatrix, mvpMatrix);
	}

	@Override
	protected void bindAttributes() {
		super.bindAttribute(0, "in_position");
		super.bindAttribute(1, "in_textureCoords");
	}

}

File: src/main/kotlin/com/adv/core/shadows/ShadowShader.kt
package com.adv.core.shadows

import com.adv.core.shaders.ShaderProgram
import org.joml.Matrix4f

class ShadowShader : ShaderProgram(VERTEX_FILE, FRAGMENT_FILE) {
    private var location_mvpMatrix = 0

    override fun getAllUniformLocations() {
        location_mvpMatrix = super.getUniformLocation("mvpMatrix")
    }

    fun loadMvpMatrix(mvpMatrix: Matrix4f) {
        super.loadMatrix(location_mvpMatrix, mvpMatrix)
    }

    override fun bindAttributes() {
        super.bindAttribute(0, "in_position")
        super.bindAttribute(1, "in_textureCoords")
    }

    companion object {
        private const val VERTEX_FILE = "shadows/shadowVertexShader.vert"
        private const val FRAGMENT_FILE = "shadows/shadowFragmentShader.frag"
    }
}
